src/Database/LSMTree/Internal/RunBuilder.hs

```diff
@@ -190,20 +203,23 @@ close fs RunBuilder {..} = do
   Helpers
 -------------------------------------------------------------------------------}
 
-writeRawPage :: HasFS IO h -> RunBuilder RealWorld (FS.Handle h) -> RawPage -> IO ()
+{-# SPECIALISE writeRawPage :: HasFS IO h -> RunBuilder (PrimState IO) (FS.Handle h) -> RawPage -> IO () #-}
+writeRawPage :: (MonadSTM m, PrimMonad m) => HasFS m h -> RunBuilder (PrimState m) (FS.Handle h) -> RawPage -> m ()
```

Comment:
Okay, I have wrapped function definitions that exceed 80 characters. My personal default is 100, so I tended to leave those type signatures in that 80 - 100 character liminal zone on one line, since it didn't look "too long" for my brain. 

Maybe we can get `stylish-haskell` to help with that?

Reply:
100 characters is also fine. As long as the lines don't get arbitrarily long (though I usually make an exception for `SPECIALISE` pragmas)